src/main/java/net/apnic/whowas/rdap/controller/RequestContext.java

```diff
@@ -0,0 +1,56 @@
+package net.apnic.whowas.rdap.controller;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Utility class for RDAP endpoint to set the current thread local context for
+ * a request.
+ */
+public class RequestContext
```

Comment:
Rather than directly deal with ThreadLocal, can we use springs abstraction instead?

```
HttpServletRequest request = ((ServletRequestAttributes) RequestContextHolder.getRequestAttributes()).getRequest();
```

This can be in turn abstracted from LinkSerializer (making it testable) with something like

```
+    @PostConstruct
    public void init()
    {
        SimpleModule module = new SimpleModule();
        Supplier<HttpServletRequest> requestSupplier = () -> ((ServletRequestAttributes) RequestContextHolder.getRequestAttributes()).getRequest()
        module.addSerializer(Link.class, new LinkSerializer(requestSupplier));
        objectMapper.registerModule(module);
    }
```

Reply:
Seems reasonable

Reply:
Done & Done